src/backend/core/api/viewsets.py

```
@@ -343,6 +348,86 @@ def stop_room_recording(self, request, pk=None):  # pylint: disable=unused-argum
+
+        serializer = serializers.RequestEntrySerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        room = self.get_object()
+        participant_id = (
+            str(request.user.id)
+            if request.user.is_authenticated
+            else request.COOKIES.get(settings.SESSION_COOKIE_NAME)
+        )
+
+        lobby_service = LobbyService()
+
+        status, livekit = lobby_service.request_entry(
+            room_id=str(room.id),
+            participant_id=participant_id,
+            user=request.user,
+            **serializer.validated_data,
+        )
+        return drf_response.Response({"status": status.value, "livekit": livekit})
+
+    @decorators.action(
+        detail=True,
+        methods=["post"],
+        url_path="enter",
+        permission_classes=[
+            permissions.HasPrivilegesOnRoom,
+        ],
+    )
+    def allow_participant_to_enter(self, request, pk=None):  # pylint: disable=unused-argument
+        """Accept or deny a participant's entry request."""
+
+        serializer = serializers.ParticipantEntrySerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+
+        room = self.get_object()
+        lobby_service = LobbyService()
+
+        try:
+            lobby_service.handle_participant_entry(
+                room_id=str(room.id),
+                **serializer.validated_data,
+            )
+            return drf_response.Response({"message": "Participant was updated."})
+
+        except LobbyParticipantNotFound:
+            return drf_response.Response(
+                {"message": "Participant not found."},
+                status=drf_status.HTTP_404_NOT_FOUND,
+            )
+        except LobbyParticipantParsingError:
+            return drf_response.Response(
+                {"message": "Participant data are corrupted, removed from the lobby."},
+                status=drf_status.HTTP_500_INTERNAL_SERVER_ERROR,
```

As you are handling the exception you should not return a 500 IMO. But... I don't find an other status code that could match this case. Maybe a 404 again ? It's like the participant does not exists as you are removing it.

---

Yes, maybe—but if the backend can't parse participant data, something might be wrong, and we'd appreciate being alerted, don't you think so?

---

In that case don't catch it. Let the exception bubble, you will have a 500 response and will be handle by a system like sentry.

---

legit